bindings/cs/MongoDB.Crypt/Library.cs

```diff
@@ -100,60 +168,124 @@ public static string Version
-        internal static readonly Delegates.mongocrypt_binary_new_from_data mongocrypt_binary_new_from_data;
-        internal static readonly Delegates.mongocrypt_binary_data mongocrypt_binary_data;
-        internal static readonly Delegates.mongocrypt_binary_len mongocrypt_binary_len;
-
-        internal static readonly Delegates.mongocrypt_ctx_new mongocrypt_ctx_new;
-        internal static readonly Delegates.mongocrypt_ctx_setopt_masterkey_aws mongocrypt_ctx_setopt_masterkey_aws;
-        internal static readonly Delegates.mongocrypt_ctx_status mongocrypt_ctx_status;
-        internal static readonly Delegates.mongocrypt_ctx_encrypt_init mongocrypt_ctx_encrypt_init;
-        internal static readonly Delegates.mongocrypt_ctx_decrypt_init mongocrypt_ctx_decrypt_init;
-        internal static readonly Delegates.mongocrypt_ctx_explicit_encrypt_init mongocrypt_ctx_explicit_encrypt_init;
-        internal static readonly Delegates.mongocrypt_ctx_explicit_decrypt_init mongocrypt_ctx_explicit_decrypt_init;
-        internal static readonly Delegates.mongocrypt_ctx_datakey_init mongocrypt_ctx_datakey_init;
-        internal static readonly Delegates.mongocrypt_ctx_setopt_masterkey_local mongocrypt_ctx_setopt_masterkey_local;
-        internal static readonly Delegates.mongocrypt_ctx_setopt_key_id mongocrypt_ctx_setopt_key_id;
-        internal static readonly Delegates.mongocrypt_ctx_setopt_key_alt_name mongocrypt_ctx_setopt_key_alt_name;
-        internal static readonly Delegates.mongocrypt_ctx_setopt_algorithm mongocrypt_ctx_setopt_algorithm;
-
-        internal static readonly Delegates.mongocrypt_ctx_state mongocrypt_ctx_state;
-        internal static readonly Delegates.mongocrypt_ctx_mongo_op mongocrypt_ctx_mongo_op;
-        internal static readonly Delegates.mongocrypt_ctx_mongo_feed mongocrypt_ctx_mongo_feed;
-        internal static readonly Delegates.mongocrypt_ctx_mongo_done mongocrypt_ctx_mongo_done;
-
-        internal static readonly Delegates.mongocrypt_ctx_next_kms_ctx mongocrypt_ctx_next_kms_ctx;
-        internal static readonly Delegates.mongocrypt_kms_ctx_endpoint mongocrypt_kms_ctx_endpoint;
-        internal static readonly Delegates.mongocrypt_kms_ctx_message mongocrypt_kms_ctx_message;
-        internal static readonly Delegates.mongocrypt_kms_ctx_bytes_needed mongocrypt_kms_ctx_bytes_needed;
-        internal static readonly Delegates.mongocrypt_kms_ctx_feed mongocrypt_kms_ctx_feed;
-        internal static readonly Delegates.mongocrypt_kms_ctx_status mongocrypt_kms_ctx_status;
-        internal static readonly Delegates.mongocrypt_ctx_kms_done mongocrypt_ctx_kms_done;
-        internal static readonly Delegates.mongocrypt_ctx_finalize mongocrypt_ctx_finalize;
-        internal static readonly Delegates.mongocrypt_ctx_destroy mongocrypt_ctx_destroy;
+        internal static Delegates.mongocrypt_version mongocrypt_version => _mongocrypt_version.Value;

```

Comment:
The properties here serve as an alias to the lazy variable's values, making it so that the caller does not need to know that the backing fields are lazy

Reply:
Ouch. So every delegate has twice as much code as before?

Reply:
Unfortunately yes: after trying out a couple of other options, this was the quickest path forward IMHO that was thread safe